### PR TITLE
[NUI.WindowSystem] Introduce the new InputGenerate feature

### PR DESCRIPTION
**Pull request by juns-kim**
## Description of Change ##
Introduce new APIs for Input generator.
The input generator can create virtual input to the Window manager.

This API Needed platform level privilege:
- http://tizen.org/privilege/inputgenerator

please refer below link the detail of input generator: 
https://developer.tizen.org/development/guides/native-application/user-interface/efl/efl-utilities#input_gen_key

### Internal APIs ###

Added:
 - public InputGenerator(DeviceType devType)
 - public InputGenerator(DeviceType devType, string name, bool sync = false)
 - public void GenerateKey(string keyName, int pressed)
 - public void GeneratePointer(int buttons, PointerType pointerType, int x, int y)
 - public void GenerateWheel(PointerWheelType wheelType, int value)
 - public void GenerateTouch(int idx, TouchType touchType, int x, int y)
 - public void GenerateTouchAxis(int idx, TouchType touchType, int x, int y, double radius_x, double radius_y, double pressure, double angle, double palm)

### CODE REVIEW

File: src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            [global::System.Runtime.InteropServices.DllImport(lib, EntryPoint = "efl_util_input_generate_wheel")]
+            internal static extern ErrorCode GenerateWheel(IntPtr inputGenHandler, int wheelType, int value);
+
+            [global::System.Runtime.InteropServices.DllImport(lib, EntryPoint = "efl_util_input_generate_touch")]
+            internal static extern ErrorCode GenerateTouch(IntPtr inputGenHandler, int idx, int touchType, int x, int y);
+
+            [global::System.Runtime.InteropServices.DllImport(lib, EntryPoint = "efl_util_input_generate_touch_axis")]
+            internal static extern ErrorCode GenerateTouchAxis(IntPtr inputGenHandler, int idx, int touchType, int x, int y, double radius_x, double radius_y, double pressure, double angle, double palm);
+
+            internal enum DeviceType
```
Comment (dongsug-song) [inline src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs:40]: could you add some comments even these are internal APIs?

File: src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                Vertical,
+                Horizontal,
+            }
+
+            private const int ErrorTzsh = -0x02860000;
+
+            internal enum ErrorCode
+            {
+                None = Tizen.Internals.Errors.ErrorCode.None,                            /* Successful */
```
Comment (dongsug-song) [inline src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs:76]: could you change c++ style comment (`/* xxx */`) to c# style (`//`)?

File: test/Tizen.NUI.WindowSystem.InputGenerator/Tizen.NUI.WindowSystem.InputGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
+﻿using System;
```
Comment (dongsug-song) [inline test/Tizen.NUI.WindowSystem.InputGenerator/Tizen.NUI.WindowSystem.InputGenerator.cs:1]: very good to add sample APP!
could you add copy right statement?
thank you.

File: src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs (hunk 1 above)
Comment (juns-kim) [reply src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs:42]: I added comments. but not detailed cause the enum name includes their information.

File: src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs (hunk 2 above)
Comment (juns-kim) [reply src/Tizen.NUI.WindowSystem/src/internal/Interop/Interop.InputGenerator.cs]: Done :smile:

File: test/Tizen.NUI.WindowSystem.InputGenerator/Tizen.NUI.WindowSystem.InputGenerator.cs (hunk 3 above)
Comment (juns-kim) [reply test/Tizen.NUI.WindowSystem.InputGenerator/Tizen.NUI.WindowSystem.InputGenerator.cs]: Done :smile:

**Review by dongsug-song — APPROVED** [review]
approved

**Pull request merged**